File: output/gpt4/java/Solution20_gpt.java


class Solution20_gpt {
    int maxSum;

    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
    
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }
    public int maxPathSum(TreeNode root) {
        maxSum = Integer.MIN_VALUE;
        getMaxPathSum(root);
        return maxSum;
    }

    private int getMaxPathSum(TreeNode node) {
        if (node == null) {
            return 0;
        }

        int leftSum = Math.max(getMaxPathSum(node.left), 0);
        int rightSum = Math.max(getMaxPathSum(node.right), 0);
        int pathSum = node.val + leftSum + rightSum;

        maxSum = Math.max(maxSum, pathSum);

        return node.val + Math.max(leftSum, rightSum);
    }




}

File: output/gpt4/js/Solution20_gpt.js
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxPathSum = function (root) {
    let maxSum = -Infinity;

    function getMaxPathSum(node) {
        if (node === null) {
            return 0;
        }

        const leftSum = Math.max(getMaxPathSum(node.left), 0);
        const rightSum = Math.max(getMaxPathSum(node.right), 0);
        const pathSum = node.val + leftSum + rightSum;

        maxSum = Math.max(maxSum, pathSum);

        return node.val + Math.max(leftSum, rightSum);
    }

    getMaxPathSum(root);
    return maxSum;
};
